atuin-server/src/router.rs

```diff
@@ -57,15 +57,20 @@ async fn teapot() -> impl IntoResponse {
     (http::StatusCode::IM_A_TEAPOT, "☕")
 }
 pub fn router(postgres: Postgres, settings: Settings) -> Router {
-    Router::new()
+    let routes = Router::new()
         .route("/", get(handlers::index))
         .route("/sync/count", get(handlers::history::count))
         .route("/sync/history", get(handlers::history::list))
         .route("/sync/calendar/:focus", get(handlers::history::calendar))
         .route("/history", post(handlers::history::add))
         .route("/user/:username", get(handlers::user::get))
         .route("/register", post(handlers::user::register))
-        .route("/login", post(handlers::user::login))
+        .route("/login", post(handlers::user::login));
+
+    let path = settings.path.as_str();
+    let valid_path = if !path.is_empty() {path} else {"/"};
+
+    Router::new().nest(valid_path, routes)
```

Comment:
I'd personally only wrap if a path is specified. 
```suggestion
    if path.is_empty() {
        routes
    } else {
        Router::new().nest(path, routes)
    }   
```